chandra_aca/star_probs.py

```diff
@@ -9,71 +9,48 @@
 from six.moves import zip
 
 from scipy.optimize import brentq
+import scipy.stats
 import numpy as np
 from Chandra.Time import DateTime
 
-# Local import in acq_success_prob():
-# from .dark_model import get_warm_fracs
-
-#
-# NOTE: the "WITH_MS" are historical and no longer used in flight
-#       and do not reflect ACA behavior beyond 2016-Feb-08.
-#
-# Scale and offset fit of polynomial to acq failures in log space.
-# Derived in the fit_sota_model_probit.ipynb IPython notebook for data
-# covering 2007-Jan-01 - 2015-July-01.  This is in the aca_stats repo.
-# (Previously in state_of_aca but moved 2016-Feb-2).
-#
-# scale = scl2 * m10**2 + scl1 * m10 + scl0, where m10 = mag - 10,
-# and likewise for offset.
+# Date of the transition from using the SOTA model to the
+# "spline" model (poly-spline-tccd) for computing star acquisition
+# probabilities.  By default the SOTA model is used for computing
+# acq probs for dates before then transition (e.g. so starcheck
+# diffs don't blow up).
+SPLINE_MODEL_TRANSITION_DATE = DateTime('2018-04-23T00:00:00')
+
+# Cache of cubic spline functions.  Eval'd only on the first time.
+SPLINE_FUNCS = {}
 
 WARM_THRESHOLD = 100  # Value (N100) used for fitting
 
-SOTA_FIT_NO_1P5_WITH_MS = [9.6887121605441173,  # scl0
-                           9.1613040261776177,  # scl1
-                           -0.41919343599067715,  # scl2
-                           -2.3829996965532048,  # off0
-                           0.54998934814773903,  # off1
-                           0.47839260691599156]  # off2
 
-SOTA_FIT_ONLY_1P5_WITH_MS = [8.541709287866361,
-                             0.44482688155644085,
-                             -3.5137852251178465,
-                             -1.3505424393223699,
-                             1.5278061271148755,
-                             0.30973569068842272]
+# Min and max star acquisition probabilities, regardless of model predictions
+MIN_ACQ_PROB = 1e-6
+MAX_ACQ_PROB = 0.985
```

Comment:
And does this only apply to the 'sota' model?  If so, should it just be defined there?

Reply:
It's used in two places (`acq_success_prob` and the SOTA model.  I decided to leave it in `acq_success_prob` for continuity with current behavior, at least for now.  Calling `spline_model_acq_prob` directly gives the raw probabilities without clipping or the empirical corrections (e.g. color=0.7).

Reply:
Ah, sorry, I thought I was searching in the full code but was accidentally grepping in the diffs.